exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -65,18 +65,20 @@ func (*clientLogger) ResponseBodyEnabled() bool {
 	return false
 }
 
-func newElasticsearchClient(logger *zap.Logger, config *Config) (*esClientCurrent, error) {
-	tlsCfg, err := config.ClientConfig.LoadTLSConfig(context.Background())
+func newElasticsearchClient(
+	ctx context.Context,
+	config *Config,
+	host component.Host,
+	telemetry component.TelemetrySettings,
+	userAgent string,
+) (*esClientCurrent, error) {
+	httpClient, err := config.ClientConfig.ToClient(ctx, host, telemetry)
 	if err != nil {
 		return nil, err
 	}
 
-	transport := newTransport(config, tlsCfg)
-
 	headers := make(http.Header)
-	for k, v := range config.Headers {
-		headers.Add(k, v)
-	}
+	headers.Set("User-Agent", userAgent)
```

Comment:
What happens to the headers in `config.Headers`? Are they used in the ES client initialization?

Reply:
They're set by the transport created by `confighttp`. See the "publish with headers" subtest of TestExporterLogs: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33367/files#diff-40c0c1ffa7ba9e2560a817797d8b388af3a57b6e63da1d62b69bc6ba09c5c8e8R152